ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/sync/PipelineChainDownloader.java

```diff
@@ -113,12 +113,6 @@ public synchronized void cancel() {
   private CompletionStage<Void> handleFailedDownload(final Throwable error) {
     pipelineErrorCounter.inc();
     final Throwable rootCause = ExceptionUtils.rootCause(error);
-    if (!cancelled.get() && rootCause instanceof CancellationException) {
```

Comment:
Is the synchronizer shutdown what is causing this failure?  If so is there a reliable test case that can induce this failure?

Reply:
Also, it looks like there are tests that depend on this being present: https://jenkins.pegasys.tech/job/Pantheon/job/PR-1260/2/testReport/junit/tech.pegasys.pantheon.ethereum.eth.sync/PipelineChainDownloaderTest/UnitTests___shouldStopWithoutCompletingFutureWhenTargetSelectionUnexpectedCancelled/

Reply:
Yes. What should we do then ?

Reply:
after reading the PR it looks like we should delete the tests.  However given that this is to ensure proper resume I think we need to spin up an AWS instance and run a fast sync, stop it a few hours in, and then see if it resumes and finishes.  We could do this against ropsten but I would prefer to see a mainnset success before we commit.

Reply:
Ok. let's do this

Reply:
I only added the test in question around the same time I added this ticket.  We should check that we can start a fast sync - wait for it to begin downloading the chain, kill pantheon and check that when it restarts the fast sync resumes.  If that works it's all good.

Testing locally, that doesn't appear to be working.  When restarting we get:
```
2019-04-15 10:51:02.965+10:00 | main | INFO  | FastDownloaderFactory | Fast sync was requested, but cannot be enabled because the local blockchain is not empty.
```

To test I ran pantheon with `./build/install/pantheon/bin/pantheon --network goerli --fast-sync-min-peers=1 --sync-mode=FAST` (after a `./gradlew installDist`)